PlanGrid.Api.Tests/AttachmentTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿// <copyright file="AttachmentTests.cs" company="PlanGrid, Inc.">
+//     Copyright (c) 2015 PlanGrid, Inc. All rights reserved.
+// </copyright>
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace PlanGrid.Api.Tests
+{
+    [TestFixture]
+    public class AttachmentTests
+    {
+        [Test]
+        public async Task UploadAttachment()
+        {
+            IPlanGridApi client = PlanGridClient.Create();
+            AttachmentUploadRequest request = await client.CreateAttachmentUploadRequest(TestData.Project2Uid, new AttachmentUpload
+            {
+                ContentType = AttachmentUpload.Pdf,
+                Name = "test name",
+                Folder = "test folder"
+            });
+
+            Stream payload = typeof(AttachmentTests).Assembly.GetManifestResourceStream("PlanGrid.Api.Tests.TestData.Sample.pdf");
+            Attachment attachment = await client.Upload(request, payload);
```

Comment:
While this is pretty easy for a client to understand, should we abstract the concept that there are two requests from end users?

e.g. `client.UploadAttachment(AttachmentUpload.Pdf, "test name", "test folder", payload);` which creates the request and puts the file.